### DISCUSSION TITLE
How could I customize a self-defined gradient function for optimize_acqf?

### DISCUSSION THREAD

**Discussion by Ksrma** [meta-pytorch/botorch#2546] [Q&A] [answered]
Hello everyone,

I have implemented a self-defined acquisition function. Its gradient also has an analytic expression, but unfortunately cannot be computed via the auto-differentiation technique in BoTorch. Then how could I provide my gradient function to optimize_acqf()? 

Thank you!

**Comment by Balandat** [comment, marked as answer]
Can you provide some more detail on the acquisition function you are considering and why auto-differentiation is not possible in this case?

The most obvious option would be to implement a custom backward pass for this acquisition function so that it can be used with existing interfaces.

**Reply by Ksrma** [reply]
Thank you a lot! I am a new user of Torch and I did not think of such a simple solution.
